src/utlls/index.ts

```diff
@@ -89,16 +91,14 @@ const prepareScalarArgument = (arg: string, dataType: AbiScalar): string | boole
 }
 
 export const prepareArguments = (
-	args: string[],
+	args: Array<string | readonly string[]>,
 	dataTypes: AbiDataType[]
 ): (string | string[] | boolean | boolean[] | BigNumber | BigNumber[])[] =>
 	args.map((arg, index) => {
-		if (dataTypes[index].endsWith("[]")) {
-			return arg
-				.split(",")
-				.map(splittedArg =>
-					prepareScalarArgument(splittedArg, dataTypes[index].slice(0, -2) as AbiScalar)
-				) as string[] | boolean[] | BigNumber[]
+		if ((dataTypes[index].endsWith("[]") && arg instanceof Array) || arg instanceof Array) {
```

Comment:
This can be simplified to `dataTypes[index].endsWith("[]") && arg instanceof Array`

Reply:
That's TS compatibility - it will complain about final return, cause if `arg instanceof Array` but `(dataTypes[index].endsWith("[]")` is false (which can't be true, we know - but TS don't know that) then TS assumes that array might drop into `prepareScalarArgument`. The best option could be to use just `arg instanceof Array`, I believe